lib/Core/Executor.cpp

```diff
@@ -1935,27 +1935,43 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
     // Arithmetic / logical
 
   case Instruction::Add: {
+    assert(i->getOperand(0)->getType()->isIntegerTy() &&
```

Comment:
No I haven't. I'm not convinced this is the right approach. If you want performance you shouldn't be running an `asserts` build. We already have lots of asserts that are expensive to check.

Do you have a particular benchmark in mind?